runtime/lua/vim/pack.lua

```diff
@@ -442,46 +428,32 @@ function PlugList:run(prepare, process, progress_title)
   for _, p in ipairs(list_noerror) do
     --- @async
     funs[#funs + 1] = function()
-      if prepare ~= nil then
-        prepare(p)
+      --- @async
+      local function plugin_f()
+        f(p)
       end
-      if #p.job.cmd == 0 or p.job.err ~= '' then
+      local function on_finish(err)
         register_finished()
```

Comment:
I suggest you inline this and suffix something like `vim.fs.basename(p.path)` to the message.

Reply:
Do you mean like adjusting `p.info.err = 'Error in ' .. p.spec.name .. ':\n' .. err`? If yes, then the data about which plugin triggered the error will be present where it is needed: either during installation or as a part of an update feedback (under plugin's name as a "title").

Reply:
Yeah, I also just noticed that predictable alignment is useful. So how about this (numbers are in parenthesis, no `:` between percent/title/message):

```
  0% Installing plugins (0/4)
 25% Installing plugins (1/4) - mini.align
 50% Installing plugins (2/4) - mini.base16
 75% Installing plugins (3/4) - mini.animate
100% Installing plugins (4/4) - mini.ai
done Installing plugins (4/4)
```